### DISCUSSION TITLE
Issue with disclosure heading

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1469] [Q&A] [answered]
Wondering if there's a known issue with the disclosure headings?

https://paste.twilio.design/components/disclosure/#composition-notes

The alignment within the heading seems to be off.

![image](https://user-images.githubusercontent.com/72031986/118844086-93eed080-b8c2-11eb-80e2-95be1b1b88ef.png)

**Comment by richbachman** [comment, marked as answer]
Hi @PoojMenon, thanks for catching this. It's probably an error in the example. We'll take a look and get it fixed.

**Reply by richbachman** [reply]
PR is up to fix this example: https://github.com/twilio-labs/paste/pull/1471

**Comment by andoyle** [comment]
Hey @richbachman I'm having a different issue with `Disclosure` headings (let me know if I should open another discussion).

I used to be able to use Flexbox within a  `Disclosure` heading to separate information using `space-between` in `Box` on its own or within `Mediabody`.
Quick mock up using the prototype kit: https://wr7s7.sse.codesandbox.io/
<img width="713" alt="Screenshot 2021-05-25 at 16 09 38" src="https://user-images.githubusercontent.com/69856100/119522769-26381e00-bd74-11eb-998d-2d5065565423.png">

I updated to the latest version of Paste lately on the prototype kit and Flex isn't working for me: https://6jzih.sse.codesandbox.io/usage
<img width="1254" alt="Screenshot 2021-05-25 at 16 23 12" src="https://user-images.githubusercontent.com/69856100/119524428-8a0f1680-bd75-11eb-8fb8-4aa0f05b10ca.png">

Is this the correct behavior now for `Disclosure` headings or is there an issue somewhere?

**Reply by richbachman** [reply]
Hi @andoyle we looked into this one, and found a potential bug in the disclosure heading. We're working on verifying its a bug and fixing. We'll update once that's done. Thanks for catching this.

**Reply by TheSisb** [reply]
Made a PR that fixes this issue https://github.com/twilio-labs/paste/pull/1487

**Reply by TheSisb** [reply]
The latest version of `@twilio-paste/core` has the fix (v9.3.5).

**Reply by andoyle** [reply]
Hey @TheSisb updated to the latest version but still not working for me on Codesandbox:
https://codesandbox.io/s/bcon-paste-components-6jzih?file=/pages/usage.js

I also updated my other sandbox that I showed it working on to see if it's something to do with my file or implementation. That's broken now too: https://codesandbox.io/s/angry-blackburn-glkce?file=/pages/index.js

**Reply by SiTaggart** [reply]
Hmmm, I do still wonder if what we had originally was the best solution where we flex-grow the child container so it never collapses? That would fix this issue for @andoyle without him doing anything to the code.

**Reply by andoyle** [reply]
Any suggestions to what I can try with my code to make it work for now? I've tried a few workarounds but nothing working for me. 

I'm only working on a proof of concept at this stage to see if I can move this page over to Paste until we get some time to completely overhaul it.

**Reply by SiTaggart** [reply]
Sorry @andoyle, this has slipped through the cracks.

Let me circle with the team, but my gut is telling me we should put it back to the way it originally was so you don't need to do anything at all. I'll either get a fix out or a solution by the end of the day.

Thanks for being patient 🙏

**Reply by andoyle** [reply]
No worries 😄

**Reply by richbachman** [reply]
Hi @andoyle 

Took a look at this more, and in order accommodate for all of the different types of elements that could be in a DisclosureHeading we feel its best to maintain the changes we made in our last fix. Unfortunately that means you'll need to make a small change in your codesandbox. You'll need to wrap the MediaObject in a Box with a 100% width. This allows the MediaObject to actually fill the entire space. Something like:

```
<Box width="100%">
    <MediaObject ... />
</Box>
```

I forked your codesandbox and added the fix: https://codesandbox.io/s/strange-flower-w9l9h?file=/pages/index.js

Apologize for the issue and the extra leg work required on this one.

**Reply by andoyle** [reply]
Ah, good auld `Box`, I should've known! That's working now. Thanks folks for looking into / helping me with this 👍
